### PR TITLE
[EasyHttpClient] Limit the metadata size in the `HttpRequestSentBreadcrumbListener` listener

### PR DESCRIPTION
**Pull request by itorgov**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: packages/EasyHttpClient/src/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListener.php [hunk 1]
```diff
@@ -7,79 +7,136 @@
...
+        return \strlen((string)\json_encode($breadcrumbData));
+    }
 
-        $this->client->leaveBreadcrumb('HTTP Request Sent', Breadcrumb::REQUEST_TYPE, $metaData);
+    /**
+     * @param array<string, mixed> $metadata
+     *
+     * @return array<string, mixed>
+     */
+    private function filterMetadata(array $metadata): array
```
Comment (akl-95) [inline packages/EasyHttpClient/src/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListener.php:106]: What abour `prepareMetadata`?

File: packages/EasyHttpClient/tests/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListenerTest.php [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+use Carbon\CarbonImmutable;
+use EonX\EasyHttpClient\Bridge\EasyBugsnag\HttpRequestSentBreadcrumbListener;
+use EonX\EasyHttpClient\Data\RequestData;
+use EonX\EasyHttpClient\Data\ResponseData;
+use EonX\EasyHttpClient\Events\HttpRequestSentEvent;
+use EonX\EasyHttpClient\Tests\AbstractTestCase;
+use Exception;
+use Mockery\MockInterface;
+
+class HttpRequestSentBreadcrumbListenerTest extends AbstractTestCase
```
Comment (akl-95) [inline packages/EasyHttpClient/tests/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListenerTest.php:18]: Please make it final.

File: packages/EasyHttpClient/tests/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListenerTest.php [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+{
+    /**
+     * @param array<string, mixed> $expectedMetadata
+     *
+     * @dataProvider provideEvents
+     */
+    public function testItSucceeds(HttpRequestSentEvent $event, array $expectedMetadata): void
```
Comment (akl-95) [inline packages/EasyHttpClient/tests/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListenerTest.php:25]: Can we follow the current approach to naming?

**Review by akl-95 — CHANGES_REQUESTED** [review]

**Review by roman-eonx — DISMISSED** [review]

File: packages/EasyHttpClient/src/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListener.php (hunk 1 above)
Comment (itorgov) [reply packages/EasyHttpClient/src/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListener.php]: Done.

File: packages/EasyHttpClient/tests/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListenerTest.php (hunk 2 above)
Comment (itorgov) [reply packages/EasyHttpClient/tests/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListenerTest.php]: Done.

File: packages/EasyHttpClient/tests/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListenerTest.php (hunk 3 above)
Comment (itorgov) [reply packages/EasyHttpClient/tests/Bridge/EasyBugsnag/HttpRequestSentBreadcrumbListenerTest.php]: Done.

**Review by akl-95 — APPROVED** [review]

**Review by natepage — APPROVED** [review]

**Pull request merged**